### ISSUE TITLE
lua errors when trying to render revealjs with fontawesome extension

### ISSUE THREAD

**Issue by morrisLuke** [quarto-ext/fontawesome#20]
Error running filter C:/Program Files/Quarto/share/filters/quarto-pre/quarto-pre.lua:
_extensions/quarto-ext/fontawesome/fontawesome.lua:56: attempt to call a nil value (field 'is_format')
stack traceback:
	...ram Files/Quarto/share/filters/quarto-pre/quarto-pre.lua:2943: in function 'callShortcodeHandler'
	...ram Files/Quarto/share/filters/quarto-pre/quarto-pre.lua:3024: in function 'transformShortcodeInlines'

reprex follows
```
---
title: "A"
subtitle: "B"
author: "C"
institute: "D"
format: 
  revealjs:
    theme:
      - moon
    self-contained: true
from: markdown+emoji
---

## [Thank you]{style="font-weight:500"}

{{< fa envelope >}} [address@email.net](mailto:address@email.net)

 

{{< fa brands github >}} [https://github.com](username)

 

{{< fa brands linkedin >}} [https://linkedin.com](handle)
```

**Comment by morrisLuke** [comment]
P.S. new to issue reporting, so let me know if anything else is needed

**Comment by cscheid** [comment]
Are you running a v1.2 prerelease version of quarto?

**Comment by mcanouil** [comment]
This is related to #19 and the issue comes from the upgrade performed based on Quarto 1.2 code change.

The solution would be to add tags/releases for the following commits:
- 63f2000b431e0af5d8b374636de965ea5ec62383 (latest version compatible with Quarto v1.1)
- HEAD (6206fc623371158bb84225803211f5fcaf73067a) (first version compatible with Quarto v1.2)

With tags, it would be possible to use:

```
quarto install extension quarto-ext/fontawesome@v0.0.1
```

@cscheid Should the extensions in https://github.com/quarto-ext be following Quarto version, at least for Major and Minor, then the patch part can be used for fixes?

**Comment by morrisLuke** [comment]
ahhh.... Looks like I've got 1.1.251, the stable on quarto's get started page.

**Comment by mcanouil** [comment]
It's fine, but the extension is currently requiring the pre-release, but the latest tag is way too old to be used.

**Comment by morrisLuke** [comment]
Just updated to the 1.2.231 prerelease and things are resolved. Thanks for tolerating a noob

**Comment by mcanouil** [comment]
@morrisLuke The extension should not be asking users to use the pre-release / not stable version without a working solution with the stable version.

**Comment by cscheid** [comment]
@mcanouil No, what should have happened is that we should update the quarto version requirement for this package.

**Comment by cscheid** [comment]
@morrisLuke Glad you got it sorted out, but no need to apologize, this is clearly a bug on the advertised version requirement.

... Oh, I see the problem now, 🤦 . The issue is that 1.1 does not have the code that checks for version requirements. That's annoying. At least we are soon going to have a stable 1.2 version, and that should solve this problem.

**Comment by mcanouil** [comment]
Meanwhile, tag/release can help @cscheid 😉

**Comment by cscheid** [comment]
@mcanouil no, it cannot, not in this case! the issue is with _quarto's_ version. Quarto 1.1 lacks the code that checks extension versions. 

Releases will not help here because we are not ever going to have the HEAD of this repo expecting an old version of quarto. The right solution would be to backport the version checking code in 1.2 back to 1.1. But that doesn't solve the issue that the current installed base of RStudio etc still will have the old, non-checking code. And the new releases of RStudio will be getting a v1.2 release anyway, like I mentioned.

**Comment by mcanouil** [comment]
The idea is to tag/release 63f2000b431e0af5d8b374636de965ea5ec62383 with for example `vO.O.2`
Then one can `quarto install extension quarto-ext/fontawesome@v0.0.2` if using Quarto < v1.2

**Comment by cscheid** [comment]
The extension support in 1.1 is poor enough that I truly would rather this not be easy to do. v1.2 is going to be out in a couple of weeks. I'm sorry, but this discussion is not particularly productive at this point.

**Comment by mcanouil** [comment]
It's fine 😉
I thought that the idea to allow to target tags when installing/using extensions/formats was for these situations among others. I am most likely missing some "long run" infos. 🤷‍♂️

**Comment by koliajaykr** [comment]
> Just updated to the 1.2.231 prerelease and things are resolved. Thanks for tolerating a noob

Please, share the code you used to resolve this issue.

**Comment by mcanouil** [comment]
He updated Quarto to v1.2 pre-release: https://quarto.org/docs/download/

**Comment by eitsupi** [comment]
Hmmm, does this mean that there are no plans to release version 1.0.0?
(I understand that quarto 1.1 does not check the required version, but isn't that unrelated to not creating a release or tag?)

**Comment by morrisLuke** [comment]
Same code as reprex, only change was downloading and installing pre-release
of Quarto CLI

**Comment by cscheid** [comment]
@eitsupi No, there will be a 1.0. I just don't want to encourage people to install an old version of quarto.

**Comment by koliajaykr** [comment]
> He updated Quarto to v1.2 pre-release: [quarto.org/docs/download](https://quarto.org/docs/download/)

Thank you, updating Quarto to the latest version (pre-release in this case) helped to solve the error related to the fontawesome.

**Comment by eitsupi** [comment]
@cscheid Now that quarto 1.2 has been released, how about creating the 1.0.0 release of this repo?

**Comment by cscheid** [comment]
@eitsupi That's a great idea.
